### PR TITLE
SQL: Use new testing framework

### PR DESCRIPTION
**Pull request by alex-spies**
Port the following test suites from `legacy-java-rest-test` to `internal-java-rest-test`:
* `x-pack/plugin/sql/qa/server/`:
  * `single-node`
  * `multi-node`
  * `multi-cluster-with-security`

Main changes:
* Update the respective `build.gradle` files.
* Add 3 new java files with the cluster definitions (as `TestRule`s), one each for `single-node`, `multi-node` and `multi-cluster-with-security`.
* Make a couple of methods non-static so they can be overridden.

A large part of the changes are mechanical.

### CODE REVIEW

File: x-pack/plugin/sql/qa/server/build.gradle [hunk 1]
```diff
@@ -38,73 +38,68 @@ subprojects {
...
+      configurations.javaRestTestRuntimeOnly {
+        // This is also required to make resolveAllDependencies work
+        resolutionStrategy.force "org.slf4j:slf4j-api:1.7.25"
+      }
+
+      /* Since we're a standalone rest test we actually get transitive
+       * dependencies but we don't really want them because they cause
+       * all kinds of trouble with the jar hell checks. So we suppress
+       * them explicitly for non-es projects. */
+      javaRestTestImplementation(project(':x-pack:plugin:sql:qa:server')) {
```
Comment (alex-spies) [inline x-pack/plugin/sql/qa/server/build.gradle:61]: Nothing below this line is new - only fixed indentation (was tripping me up, haha).

**Comment by alex-spies** [comment]
@elasticsearchmachine run Check labels

File: x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/SqlSecurityTestCase.java [hunk 2]
```diff
@@ -56,6 +56,13 @@
 import static org.hamcrest.Matchers.is;
 
 public abstract class SqlSecurityTestCase extends ESRestTestCase {
+    public String elasticsearchAddress() {
+        // CLI only supports a single node at a time so we just give it one.
+        // String cluster = System.getProperty("tests.rest.cluster");
+        // return cluster.split(",")[0];
```
Comment (luigidellaquila) [inline x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/SqlSecurityTestCase.java:62]: Leftover?

**Review by luigidellaquila — APPROVED** [review]
For my understanding it looks good, thanks Alex!
I just added a very minor comment

File: x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/SqlSecurityTestCase.java (hunk 2 above)
Comment (alex-spies) [reply x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/SqlSecurityTestCase.java]: Ah, I was so sure I had this already removed. Thanks, removed!

**Review by breskeby — COMMENTED** [review]
lgtm. 
FYI: I've updated the whitelist we keep in our build and removed those projects that have been converted here too https://github.com/elastic/elasticsearch/pull/102297

**Pull request merged**